aiida/tools/archive/create.py

```diff
@@ -259,7 +259,7 @@ def create_archive(
     if test_run:
         EXPORT_LOGGER.report('Test Run: Stopping before archive creation')
         keys = set(
-            orm.Node.collection(backend).iter_repo_keys(
+            orm.nodes.node.NodeCollection(orm.Node, backend).iter_repo_keys(
```

Comment:
Was this causing a bug? If I look at the `Collection` constructor, if the base entity type is not explicitly defined, it would take the `_entity_base_cls` which is defined for `Node`. So why was this not working? Maybe the change should have simply been:
```suggestion
            orm.nodes.node.NodeCollection(backend=backend).iter_repo_keys(
```
since `backend` was being misinterpreted for as `entity_class` since it was the first and only positional argument?

Reply:
As a complete aside (and not for this PR), we probably should probably change `Collection.__init__` to:
```python
class Collection(abc.ABC, Generic[EntityType]):

    def __init__(self, backend: Optional['StorageBackend'] = None, entity_class: Type[EntityType]) -> None:
        """ Construct a new entity collection.

        :param backend: the backend instance to get the collection for, or use the default
        :param entity_class: the entity type e.g. User, Computer, etc
        """
        from aiida.orm.implementation import StorageBackend

        self._entity_type = entity_class or self._entity_base_cls()
        assert issubclass(self._entity_type, self._entity_base_cls())

        self._backend = backend or get_manager().get_profile_storage()
        type_check(backend, StorageBackend, allow_none=True)
```
that is to say to swap `backend `and `entity_class`. Usually the entity cls will be defined in the subclass through the `_entity_base_cls` classmethod and it is the backend that we want to specify. So it makes more sense to have that as the first positional argument.

Reply:
So `orm.nodes.node.NodeCollection(backend=backend)` will definitely not work at present, because `NodeCollection` does not implement its own `__init__` and `Collection.__init__` explicitly requires the `entity_class`.

Your change indeed makes sense, that the `entity_class` should not be needed to initialise 👍 

and yeh the whole `orm.Node.collection(backend)` API, I feel is a little "clunky",
because I don't like that it is having to "silently" load the global backend first (which for a temp backend may not have even been loaded).

Reply:
I don't understand how any of this is working properly now. You are right that `entity_class` is currently required in the `Collection` constructor, but then how are things like
```
class SqliteTempBackend:
    @cached_property
    def authinfos(self):
        return orm.SqliteAuthInfoCollection(self)
```
correct? The `SqliteAuthInfoCollection` is instantiated passing the `SqliteTempBackend` instance as the first argument, i.e., for `entity_class`, but the `SqliteAuthInfoCollection` would expect a `BackendAuthInfo` as per its `entity_cls` implementation? It seems that this code is trying to define the `backend` by passing itself, but it should be misinterpreted for the `entity_class`? If that is true I would expect a whole load of bugs which is clearly not happening so what is going on here?

Reply:
Ah never mind, that collection actually references a `BackendCollection` which actually only takes `backend` as an argument. Ok, that should really be the way the front-end `Collection` works as well. Anyway, we can ignore this discussion for this PR as it is not relevant. I will have a look at simplifying the `Collection` implementation